src/hooks/coingecko/fetchFromCoinGecko.ts

```diff
@@ -1,12 +1,17 @@
-export const fetchFromCoinGecko = async (currency: string, name: string): Promise<number> => {
+interface CoinGeckoResponse {
+  [name: string]: {
+    [key: string]: number
+  }
+}
+
+export const fetchFromCoinGecko = async (currency: string, names: string): Promise<CoinGeckoResponse> => {
   const response = await fetch(
-    `https://api.coingecko.com/api/v3/simple/price?ids=${name}&vs_currencies=${currency}`,
+    `https://api.coingecko.com/api/v3/simple/price?include_market_cap=true&include_24hr_vol=true&include_last_updated_at=true&precision=full&ids=${names}&vs_currencies=${currency.toUpperCase()}`,
```

Comment:
If you're not using the last updated time you could remove it from the request here. 
```suggestion
    `https://api.coingecko.com/api/v3/simple/price?include_market_cap=true&include_24hr_vol=true&precision=full&ids=${names}&vs_currencies=${currency.toUpperCase()}`,
```